test/expr/unary/selector/selector.swift

```diff
@@ -146,3 +146,19 @@ case #selector(SR1827.bar)?:
 default:
   break
 }
+
+// SR-3174
+
+class Hello {
```

Comment:
I added a test for a nested declaration. What exactly do you mean by "top-level selector"?